etc/docker/elyra/Dockerfile

```diff
@@ -32,7 +32,8 @@ RUN chmod ugo+x /usr/local/bin/start-elyra.sh && \
 USER $NB_USER
 
 RUN python -m pip install --upgrade pip && \
-    python -m pip install setuptools pandas --ignore-installed --upgrade && \
+    python -m pip install --ignore-installed --upgrade setuptools && \
+    python -m pip install --upgrade pandas numpy papermill nbclient jupyter-client notebook  && \
```

Comment:
The upgrade references of `papermill nbclient jupyter-client notebook` are redundant with what `make install` does below.  Is this because  we aren't able to specify `--upgrade-strategy eager` in `make install`?  Would it be better to add a macro and call something like `make UPGRADE_STRATEGY=eager install` and update the Makefile `install-server` target to appropriately use `--upgrade-strategy` as follows...
```
UPGRADE_STRATEGY?=only-if-needed

...

install-server: build-server ## Install backend
	pip install --upgrade --upgrade-strategy $(UPGRADE_STRATEGY) dist/elyra-*-py3-none-any.whl
```

Reply:
These are the papermill transient dependencies that always bite us, and thus why I added them explicitly.

Reply:
Understood, but we list most of those dependencies as part of [Elyra's installation requirements](https://github.com/elyra-ai/elyra/blob/master/setup.py#L50-L68) and an eager upgrade would address the others - thus my comment about redundancy.  Should we ever need to pin (cap) one of these dependencies, the approach in the PR will potentially break the container at runtime since a module can be installed that is greater than it's cap.

I'm OK moving forward without this but wanted to note the reason I'm bringing this up in the first place.